Stop destroying panadapters

nsbruce opened the pull request:
It turns out #16 was created without understanding the problem. Panadapters were not being destroyed. In order to free up the flex for multiple pan support the included fixes DOES destroy all pans on the flex.

nsbruce commented:
These latest commits can be included too @cjam

cjam commented on python/flex.py:
```diff
@@ -46,22 +53,32 @@ def __initRadio(self):
+                    destroyed_pans += 1
+                pan_check = FlexApi.radio.WaitForPanadaptersSync()
+        try:
+            print("{0} panadapters destroyed".format((destroyed_pans - 1) / 2))
```
Do we need this print statement in a try catch?

cjam replied:
Ah I see what's happened here.  It's because `destroyed_pans` may not be defined at this point

cjam replied:
Ok, I wonder if we could make things a little bit more succinct:

```python
        print("flex::WaitForPanadaptersSync")  # Log what we do before, since the syncrhonous call below will block
        existing_pans = FlexApi.radio.WaitForPanadaptersSync()

        if len(existing_pans) == 0:
           print("No panadapters active.")
        else:
             print("Destroying panadapters...")
             destroyed_pans = 0
             while len(existing_pans) is not 0:
                 for p in existing_pans:
                     p.Close(True)
                     destroyed_pans += 1
                 existing_pans = FlexApi.radio.WaitForPanadaptersSync()
             print("{0} panadapters destroyed".format((destroyed_pans - 1) / 2))
```

Notice that you no longer need the `try/catch` because destroyed_pans is always defined in that scope.  Looking at it, do you still need to do the loop?  Could you use the new parameter that we added into the WaitForPanadaptersSync? i.e. `FlexApi.radio.WaitForPanadaptersSync(2000,1000) # wait a 1 second before returning panadapters list`

nsbruce commented:
I simplified as you suggested. I just got rid of destroyed_pans altogether. I was being overly verbose to debug.

When I replaced the whole segment with:
```
        print("flex::WaitForPanadaptersSync")
        existing_pans = FlexApi.radio.WaitForPanadaptersSync(2000, 1000)
        print("Number of active panadapters: {0}".format(
            len(existing_pans)))
        for p in existing_pans:
            p.Close(True)
```
I get the following failure. It seems to be something pythonnet related (see pythonnet/pythonnet/#226)
```
flex::WaitForPanadaptersSync
handler caught exception: No method matches given arguments
Traceback (most recent call last):
  File "/home/nsbruce/pybombs-prefix/GR-for-FlexRadio/lib/python2.7/dist-packages/gnuradio/gr/gateway.py", line 55, in eval
    try: self._callback()
  File "/home/nsbruce/pybombs-prefix/GR-for-FlexRadio/lib/python2.7/dist-packages/gnuradio/gr/gateway.py", line 170, in __gr_block_handle
    self.__message.start_args_return_value = self.start()
  File "/home/nsbruce/pybombs-prefix/GR-for-FlexRadio/lib/python2.7/dist-packages/flex/FlexSource.py", line 154, in start
    self.radio = FlexApi().getRadio()
  File "/home/nsbruce/pybombs-prefix/GR-for-FlexRadio/lib/python2.7/dist-packages/flex/flex.py", line 85, in getRadio
    self.__initRadio()
  File "/home/nsbruce/pybombs-prefix/GR-for-FlexRadio/lib/python2.7/dist-packages/flex/flex.py", line 77, in __initRadio
    existing_pans = FlexApi.radio.WaitForPanadaptersSync(2000, 1000)
TypeError: No method matches given arguments
thread[thread-per-block[0]: <block source (3)>]: caught unrecognized exception
```

nsbruce commented:
Okay I think that does it @cjam

cjam commented on python/flex.py:
```diff
@@ -23,33 +23,32 @@
     print("Loaded Assembly: {0}".format(name))
 
-# Import the API
+# Import the API (has to be after the loaded assemblies are printed)
```
It just needs to be after the clr.AddReference calls.  

I'm only printing out loaded assemblies for debugging purposes.

nsbruce replied:
fixed in latest commit

cjam commented on python/flex.py:
```diff
@@ -58,25 +57,20 @@ def __initRadio(self):
-            print("Destroying panadapters...")
-            while len(pan_check) is not 0:
-                for p in pan_check:
+        existing_pans = FlexApi.radio.WaitForPanadaptersSync()
```
This looks more succinct for sure.

cjam commented:
Nick, About the error that your getting.  Can you double check what your FlexLibMono radio.WaitForPanadaptersSync looks like.  I just want to make sure that your branch has those latest updates.

nsbruce commented:
@cjam Yea everything is up to date. I could just hard set the delay longer in Radio.cs but that feels like a huge cop-out. I wonder if the pythonnet error is stopping something from loading those changes though? I'll fiddle on Monday. We can consider it a future finicky enhancement.
